llama_hub/tools/chatgpt_plugin/base.py

```diff
@@ -0,0 +1,54 @@
+"""ChatGPT Plugiun Tool."""
+
+from typing import List
+
+from llama_index.readers.schema.base import Document
+from llama_index.tools.tool_spec.base import BaseToolSpec
+from llama_hub.tools.openapi.base import OpenAPIToolSpec
+from llama_hub.tools.requests.base import RequestsToolSpec
+from llama_index.tools.types import ToolMetadata
+import requests
+import yaml
+
+class ChatGPTPluginToolSpec(BaseToolSpec):
+    """ChatGPT Plugin Tool
+
+    This tool leverages the OpenAPI tool spec to automatically load ChatGPT 
+    plugins from a manifest file
+    """
+
+    spec_functions = ['describe_plugin']
```

Comment:
hmm i think the tool spec interface is a bit confusing now that we're overriding `to_tool_list`. a bit hard to tell now which tools are actually getting returned to the agent, and the `spec_functions` don't quite match anymore

Reply:
do you have thoughts on how to simplify this?

Reply:
also, unrelated maybe to this PR but out of curiosity why does the openapi tool take in a pre-existing dict instead of loading the openapi spec as part of the function

Reply:
Yeah I see what you mean, I think if we wanted to make the interface more intuitive from the code we could:

- Require the user to import and pass the Requests tool spec, similar to what OpenAPI tool spec requires right now.
- have ChatGPT plugin export its own load_openapi_spec that calls the internally wrapped OpenAPIToolSpec, and add it to spec functions

With that we wouldn't need to override to_tool_list and make it potentially a bit easier for users to customize request headers. I think this seems a bit more straight forward, I'll put together some code.

Reply:
For the openapi tool taking a spec, my thinking was it allows users to load a spec from url or local file, but we could also update the interface to accept a file path or url and load the spec from there. I can't really imagine users wanting to load an OpenAPI spec in a different way.

Reply:
yeah that's more intuitive, it unbundles loading the spec from actually calling the spec (with the requests tool). We can just treat the chatgpt plugin as a special case of the openapi spec